pandas/core/indexes/base.py

```diff
@@ -483,6 +483,26 @@ def _simple_new(cls, values, name: Label = None):
     def _constructor(self):
         return type(self)
 
+    def _maybe_check_unique(self):
```

Comment:
where is this called? is it public?

Reply:
Not public. It's called whenever `.flags.allows_duplicate_labels` is set to True (so in `.set_flags(...)` or via `.__finalize__`.